libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -391,20 +401,56 @@ def displayStatus(self):
         return self.crc.result.state == 'canceled'
 
+    def _set_boot_iso_path(self):
+        """ Sets path to boot iso for virt-install command """
+        if self.hypervisor.remote:
+            # TODO: Add support for boot iso with remote hypervisors.
+            # Proposed solution: add functionality to the Hypervisor class to copy or download the boot iso to the hypervisor
+            raise ResourceNotAvailable('anaconda-webui needs InstallationSource structure when the VM hypervisor is remote')
+
+        if self.boot_iso_structure[self.architecture].startswith(('http://', 'https://')):
+            self.boot_iso_path = os.path.join(self.temp_dir, f'boot-{self.architecture}.iso')
+            if os.path.exists(self.boot_iso_path):
+                LOGGER.info(f'Using existing iso: {self.boot_iso_path}')
+                return
+
+            # Download boot iso
+            LOGGER.info(f'Downloading {self.boot_iso_structure[self.architecture]}')
+            self.boot_iso_path, _ = urllib.request.urlretrieve(
+                self.boot_iso_structure[self.architecture],
+                self.boot_iso_path
+            )
+
+        elif self.boot_iso_structure[self.architecture].startswith('file://'):
+            # Local boot iso, file:// url
+            self.boot_iso_path = self.boot_iso_structure[self.architecture][7:]
+        else:
+            # Local boot iso
+            self.boot_iso_path = self.boot_iso_structure[self.architecture]
+
+        if not os.path.exists(self.boot_iso_path):
+            raise FileNotFoundError(self.boot_iso_path)
+
     def _start_vm(self):
         """ Prepare virt-install command and start VM """
         self.log('Starting VM', show=True)
 
-        # Get compose location
-        os_url = self.installation_source.base_repo[self.architecture]['os']
-        if (self.installation_source.kernel_path(self.architecture) and
-            self.installation_source.initrd_path(self.architecture)):
-            location = f'{os_url},kernel={self.installation_source.kernel_path(self.architecture)},initrd={self.installation_source.initrd_path(self.architecture)}'
+        # Construct kernel cmdline for --extra-args
+        kernel_cmdline = f'inst.sshd inst.webui inst.webui.remote inst.graphical console=ttyS0'
+
+        # Get installation source location
+        if self.boot_iso_path is None:
+            os_url = self.installation_source.base_repo[self.architecture]['os']
+            if (self.installation_source.kernel_path(self.architecture) and
+                self.installation_source.initrd_path(self.architecture)):
+                location = f'{os_url},kernel={self.installation_source.kernel_path(self.architecture)},initrd={self.installation_source.initrd_path(self.architecture)}'
+            else:
+                location = os_url
+
+            kernel_cmdline += f' inst.stage2={os_url}'
         else:
-            location = os_url
+            location = self.boot_iso_path
```

Comment:
I am afraid we may need to set path to kernel and initrd also for boot.iso. There has been a change of location in the iso images quite recently and it seems libvirt is not able to find the kernel on default place on some of our runners (F34). Maybe we could use InstallationStructure to supply the path via its kernel and initrd settings but I am not sure if it is clever, confusing, or dirty soulution.

Other option would be to require runners >= F37. I guess default location should work on F37 and newer.

Reply:
I don't like the idea of adding workarounds for EOL versions of Fedora. The runners should be upgraded. Or if that would be a problem maybe we could limit what runners are used for this workflow?